grafana/resource_team_external_group.go

```diff
@@ -0,0 +1,164 @@
+package grafana
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceTeamExternalGroup() *schema.Resource {
+	return &schema.Resource{
+
+		Description: `
+* [Official documentation](https://grafana.com/docs/grafana/latest/enterprise/team-sync/)
+* [HTTP API](https://grafana.com/docs/grafana/latest/http_api/external_group_sync/)
+`,
+
+		CreateContext: CreateTeamExternalGroup,
+		UpdateContext: UpdateTeamExternalGroup,
+		DeleteContext: DeleteTeamExternalGroup,
+		ReadContext:   ReadTeamExternalGroup,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"team_id": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The Team ID",
+			},
+
+			"groups": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "The team external groups list",
+			},
+		},
+	}
+}
+
+func CreateTeamExternalGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	teamID := d.Get("team_id").(int)
+	d.SetId(strconv.FormatInt(int64(teamID), 10))
+	if err := UpdateGroups(d, meta); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diag.Diagnostics{}
+}
+
+func ReadTeamExternalGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if err := ReadGroups(d, meta); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
```

Comment:
```suggestion
	return diag.Diagnostics{}
```
Trivial, but I suppose for the sake of consistency.